cpp/src/binaryop/compiled/operation.cuh

```diff
@@ -388,9 +388,10 @@ struct ATan2 {
 
 struct NullEquals {
   template <typename TypeLhs, typename TypeRhs>
-  CUDA_DEVICE_CALLABLE auto operator()(TypeLhs x, TypeRhs y, bool lhs_valid, bool rhs_valid)
-    -> decltype(x == y)
+  CUDA_DEVICE_CALLABLE auto operator()(
+    TypeLhs x, TypeRhs y, bool lhs_valid, bool rhs_valid, bool& output_valid) -> decltype(x == y)
```

Comment:
Use `thrust::optional` here instead?

Reply:
For `bool& output_valid`? why?
`bool& output_valid` is added to `NullEquals` to make it similar `NullMax` and `NullMin`. It's easier for constexpr conditions.
`output_valid` is infact won't be used.

Reply:
For `output_valid` and to replace `lhs_valid` and `rhs_valid`. It's far more clear and intuitive than having separate parameters.

Reply:
```
thrust::optional<bool> operator()(thrust::optional<LHS> lhs, thrust::optional<RHS> rhs)
```

Reply:
So far when I tried to implement it, it seems to complicate `constexpr` and also the operator NullEquals, NullMax, NullMin.
This will also complicate `is_binary_operation_supported` constexpr and increase its compiled time.

I couldn't get NullMax and NullMin to work.
`std::is_invocable_v<BinaryOperator, thrust::optional<TypeLhs>, thrust::optional<TypeRhs>>` doesn't return true for NullMax and NullMin in ops_wrapper and op2_wrapper.
but it returns true at `is_binary_operation_supported`.